src/tvp/controller.ts

```diff
@@ -0,0 +1,79 @@
+import { commands, workspace, Disposable } from "coc.nvim"
+
+import { TreeModel } from "./model"
+import { TreeViewFeature } from "./feature"
+import { TreeViewsManager, WindowProp } from "./treeviews"
+
+export class TreeViewController implements Disposable {
+  private treeModels: Map<string, TreeModel> = new Map()
+  private listeners: Disposable[] = []
+
+  public constructor(
+    treeViewFeature: TreeViewFeature,
+    private treeViewsManager: TreeViewsManager
+  ) {
+    this.listeners.push(
+      treeViewFeature.providerEvents()(provider => {
+        const model = new TreeModel(provider)
+        this.treeModels.set(provider.viewId, model)
+        treeViewsManager.addTreeModel(model)
+      })
+    )
+
+    this.listeners.push(
+      commands.registerCommand("metals.tvp", viewName => {
```

Comment:
Can we be more explicit with `viewName` here, especially since this will only ever be `metalsBuild` or `metalsCompile`. Maybe an Enum?

Reply:
First of all let me say that code inside directory src/tvp doesn't depend on `metalsBuild` or `metalsCompile`. Metals can add new views or remove existing one but it should work without any changes in coc-metals. User should just update hotkeys in .vimrc.

I don't understand what we can get here by adding enum.

Reply:
Ah, I misunderstood then. Sounds good.